### PR TITLE
build: upgrade FOCA to 0.13.0

### LINKED ISSUE #49: Upgrade FOCA to latest version
Upgrade FOCA to version [0.13.0](https://github.com/elixir-cloud-aai/foca/releases/tag/v0.13.0)

### PR DESCRIPTION
**Pull request by jvkersch**
## Description

Updates FOCA to the latest version, which is 0.13.0 at the time this PR was opened.

Fixes #49 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [x] My code follows the [style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/contributing_guidelines.md#language-specific-guidelines) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage
- [ ] I have added docstrings following the [Python style guidelines](https://github.com/elixir-cloud-aai/elixir-cloud-aai/blob/dev/resources/python.md) of this project to all new modules, classes, methods and functions are documented with docstrings following; I have updated any previously existing docstrings, if applicable
- [ ] I have updated any sections of the app's documentation that are affected by the proposed changes, if applicable

## Summary by Sourcery

This pull request updates the FOCA library to version 0.13.0, refactors the code to use the new configuration syntax, and adjusts the test cases accordingly to ensure compatibility and functionality.

* **Bug Fixes**:
    - Updated FOCA to version 0.13.0 to fix compatibility issues.
* **Enhancements**:
    - Refactored test cases to use the updated FOCA configuration syntax.
* **Tests**:
    - Modified test cases to align with the new FOCA configuration structure.

### CODE REVIEW

File: drs_filer/ga4gh/drs/endpoints/register_objects.py [hunk 1]
```diff
@@ -32,24 +32,24 @@ def register_object(
     """
     # Set parameters
     db_collection = (
-        current_app.config['FOCA'].db.dbs['drsStore'].
+        current_app.config.foca.db.dbs['drsStore'].
```
[the remark this answers is not in the document]
Comment (kushagra189) [reply drs_filer/ga4gh/drs/endpoints/register_objects.py]: +1 to this

**Review by kushagra189 — DISMISSED** [review]
Only one minor comment. We have made some repetitive usages but we can take that up in a separate refactor as well.

File: drs_filer/ga4gh/drs/endpoints/register_objects.py [hunk 2]
```diff
@@ -121,10 +121,10 @@ def __add_access_ids(data: List) -> List:
         Access methods metadata complete with unique access identifiers.
     """
     id_charset = eval(
-        current_app.config['FOCA'].endpoints['access_methods']['id_charset']
+        current_app.config.foca.endpoints['access_methods']['id_charset']
```
[the remark this answers is not in the document]
Comment (jvkersch) [reply drs_filer/ga4gh/drs/endpoints/register_objects.py]: Flagged as https://github.com/elixir-cloud-aai/drs-filer/issues/54.

File: drs_filer/ga4gh/drs/endpoints/register_objects.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jvkersch) [reply drs_filer/ga4gh/drs/endpoints/register_objects.py]: Done!

File: drs_filer/ga4gh/drs/endpoints/register_objects.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jvkersch) [reply drs_filer/ga4gh/drs/endpoints/register_objects.py]: I can address #54 right after this PR, but I prefer not mixing maintenance changes and behavioral changes.

File: drs_filer/ga4gh/drs/server.py [hunk 3]
```diff
@@ -34,7 +34,7 @@ def GetObject(object_id: str) -> Dict:
         DRS object as dictionary, JSONified if returned in app context.
     """
     db_collection = (
-        current_app.config['FOCA'].db.dbs['drsStore'].
+        current_app.config.foca.db.dbs['drsStore'].
```
[the remark this answers is not in the document]
Comment (jvkersch) [reply drs_filer/ga4gh/drs/server.py:37]: Not applicable; the config object is referred to in different functions.

**Comment by jvkersch** [comment]
Thanks @kushagra189, I've done that particular refactor in the module where there were a lot of repeated usages of that object. There was another instance where Sourcery suggested a refactor but that was not applicable (the repeated code was spread out over multiple functions). 

The only open issue is the use of `eval` (#54). I would suggest merging this PR as-is and tackling that issue separately.

**Review by kushagra189 — APPROVED** [review]
Thanks @jvkersch , LGTM.

**Pull request merged**